### PR TITLE
Migration of Process Module

### PR DESCRIPTION
**Pull request by harvey0100**
Migration of Process Module
Improved Unit/Functional / RST Docstrings

Reference: https://github.com/avocado-framework/aautils/issues/52

Somehow closed the previous PR must have misclicked.

## Summary by CodeRabbit

* **New Features**
  * Adds a comprehensive process and command-execution utility with robust I/O, timeout, sudo/capability checks, background handling, and rich result reporting.

* **Documentation**
  * Adds developer docs referencing the new process utilities.

* **Tests**
  * Adds extensive unit and functional tests validating subprocess behavior, timeouts, encoding, environment propagation, large output, and edge cases.

* **Chores**
  * Adds metadata descriptor and test helper utilities for logging and conditional test skipping.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: autils/devel/process.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def can_sudo(cmd=None):
     try:
```
Comment (richtja) [inline autils/devel/process.py:95]: Maybe I am missing something here, but IMO this commit can be merged to the previous one since we are introducing the new utility here we don't need to have these changes separated.

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @harvey0100, thank you very much for the changes. IMO it LGTM I have just one more proposal, please take a look.

File: autils/devel/process.py (hunk 1 above)
Comment (harvey0100) [reply autils/devel/process.py:95]: Updated now :)

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @harvey0100, IMO the CodeRabbit is right here about update of the paths in the docstrings from `avocado.utils` to `autils`. Can you please make this one more change? Thank you.

**Comment by harvey0100** [comment]
> Hi @harvey0100, IMO the CodeRabbit is right here about update of the paths in the docstrings from `avocado.utils` to `autils`. Can you please make this one more change? Thank you.

Ah certainly missed out on this comment, I've got this sorted now in the latest commit.

**Review by richtja — APPROVED** [review]
LGTM
